crates/asm/subprotocols/core/src/parsing/checkpoint.rs

```diff
@@ -21,33 +21,7 @@ pub(crate) fn extract_signed_checkpoint(tx: &TxInputRef<'_>) -> Result<SignedChe
     // a mock TxFilterConfig.
     let filter_config = mock_checkpoint_filter_config();
 
-    // Parse checkpoint envelopes using the same pattern as strata-l1tx
-    let checkpoints: Vec<SignedCheckpoint> = tx
-        .tx()
-        .input
-        .iter()
-        .flat_map(|inp| {
-            inp.witness
-                .taproot_leaf_script()
-                .and_then(|scr| parse_envelope_payloads(&scr.script.into(), &filter_config).ok())
-                .map(|items| {
-                    items
-                        .into_iter()
-                        .filter_map(|item| match *item.payload_type() {
-                            L1PayloadType::Checkpoint => {
-                                // Deserialize checkpoint from payload data
-                                borsh::from_slice(item.data()).ok()
-                            }
-                            L1PayloadType::Da => None,
-                        })
-                        .collect::<Vec<_>>()
-                })
-                .unwrap_or_default()
-        })
-        .collect();
-
-    // Return the first valid checkpoint found
-    checkpoints.into_iter().next().ok_or_else(|| {
+    parse_valid_checkpoint_envelope(tx.tx(), &filter_config).ok_or_else(|| {
```

Comment:
i'll adapt similar stuff in my PR for the `subprotocols/checkpoint-v0`, that's helpful as it is, thank you!